src/hooks/faktum/useValidateFaktumDato.ts

```diff
@@ -0,0 +1,81 @@
+import { isFuture } from "date-fns";
+import { useState } from "react";
+import {
+  isFromYear1900,
+  isOverTwoWeeks,
+  isWithinValidYearRange,
+} from "../../components/faktum/validation/validations.utils";
+import { useSanity } from "../../context/sanity-context";
+import { useValidation } from "../../context/validation-context";
+import { QuizFaktum } from "../../types/quiz.types";
+
+type dateFaktumErrorType = "invalidDate" | "invalidBirthDate";
+
+interface IProps {
+  getErrorMessage: () => string | undefined;
+  getWarningMessage: () => string | undefined;
+  isValidDate: (value: Date) => boolean;
+}
+
+export function useValidateFaktumDato(faktum: QuizFaktum): IProps {
+  const { getAppText, getFaktumTextById } = useSanity();
+  const { unansweredFaktumId } = useValidation();
+  const faktumTexts = getFaktumTextById(faktum.beskrivendeId);
+  const [hasError, setHasError] = useState<boolean | dateFaktumErrorType>(false);
```

Comment:
Det ser ut som den boolske verdien til hasError bare brukes for å nullstille feil? I så fall er det kanskje bedre å sette typen til const [hasError, setHasError] = useState<dateFaktumErrorType | undefined>();

Reply:
Godt poengt, jeg er veldig enig med deg her.